src/gdpc/minecraft_tools.py

```diff
@@ -211,125 +2521,171 @@ def bookData(
+                    pages.append(page)
+                lineNumber = 1
+                pageStartIndex = wordStartIndex
+                lastWordEndIndex = wordStartIndex
+            else:
+                lineNumber = lineNumber + 1
+            lineColumnCount = wordColumnCount
         else:
-            page = [[word for word in line.split()] for line in page.split('\n')]
-        for line in page:
-            toprint = ""
-            for word in line:
-                if pixels_left != _BOOK_PIXELS_PER_LINE:
-                    if characters_left < 1:
-                        newpage()
-                    elif SPACE_WIDTH > pixels_left:
-                        newline()
-                    else:
-                        toprint += ' '
-                        characters_left -= 1
-                        pixels_left -= SPACE_WIDTH
-
-                width = fontwidth(word)
-                if width > pixels_left:
-                    if width > _BOOK_PIXELS_PER_LINE:  # cut word to fit
-                        original = word
-                        for letter in original:
-                            charwidth = fontwidth(letter) + 1
-                            if charwidth > pixels_left:
-                                newline()
-                            toprint += letter
-                            width -= charwidth
-                            word = word[1:]
-                            characters_left -= 1
-                            pixels_left -= charwidth
-                            if not width > pixels_left:
-                                break
-                    else:
-                        newline()
-                if len(word) > characters_left:
-                    newpage()
-                toprint += word
-                characters_left -= len(word)
-                pixels_left -= width
-            newline()           # finish line
-        newpage()               # finish page
-    del outputPages[-1] # end last page (book is complete)
-
-    loreJSON = json.dumps({"text": description, "color": desccolor, "italic": descIsItalic})
-    pageJSON = [json.dumps({"text": p}) for p in outputPages]
-    return (
+            lineColumnCount = nextLineColumnCount
+        cursor = cursor + 1
+
+    loreJSON = json.dumps({'text': description, 'color': descriptionColor, 'italic': descriptionIsItalic})
+    pageJSON = [json.dumps({'text': p}) for p in pages]
+    book = (
         "{"
-            "\"minecraft:written_book_content\": {"
-                f'title: {repr(title)}, '
-                f'author: {repr(author)}, '
-                f'pages: [{",".join(repr(p) for p in pageJSON)}]'
-            "},"
-            f'"lore": [{repr(loreJSON)}]'
+        "\"minecraft:written_book_content\": {"
+        f'title: {repr(title)}, '
+        f'author: {repr(author)}, '
+        f'pages: [{",".join(repr(p) for p in pageJSON)}]'
+        "},"
+        f'"lore": [{repr(loreJSON)}]'
```

Comment:
Please revert, I like the indentation here for clarity.
```suggestion
            "\"minecraft:written_book_content\": {"
                f'title: {repr(title)}, '
                f'author: {repr(author)}, '
                f'pages: [{",".join(repr(p) for p in pageJSON)}]'
            "},"
            f'"lore": [{repr(loreJSON)}]'
```

(I'll add format-ignore comments for these cases if/when I add an autoformatter)